src/mscorlib/src/System/String.cs

```diff
@@ -1286,10 +1286,39 @@ internal String[] SplitInternal(char[] separator, int count, StringSplitOptions
 
             return result;
         }
-    
-    
-        // Removes a string of characters from the ends of this string.
+
+
+        // Overloads the Trim functions (Trim, TrimEnd, Trim Start)
+        // Removes a string of characters from the ends of this string,
+        // But accepts a single char instead of an array, reducing memory use - added 11/14/15
         [Pure]
+        public String Trim(char trimChars)
+        {
+            if (trimChars == '\0')
```

Comment:
I don't think this is a good idea.  For the Trim(char[]) it makes sense to pass in null as a way to get the default set of things to Trim, but I think if we pass the '\0' character to the single argument Trim we should actually just Trim all instances of '\0' but leave everything else.